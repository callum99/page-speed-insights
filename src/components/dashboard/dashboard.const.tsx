import {currentPageType, historicPagesType} from "../global/global.types";

export const singlePageMock:currentPageType = {
    performance: {
        TTFB: "0.1",
        FullLoadTime: "0.1",
        DOMContentLoadedTime: "0.1"
    }
};

export const historicPageMock:historicPagesType = {
    ["2024-04-24T10:52:20.779Z"]: {
        ...singlePageMock
    },
    ["2024-04-24T10:12:09.664Z"]: {
        ...singlePageMock
    },
    ["2024-04-24T09:34:38.019Z"]: {
        ...singlePageMock
    },
    ["2024-04-24T09:10:46.905Z"]: {
        ...singlePageMock
    },
    ["2024-04-24T08:13:14.099Z"]: {
        ...singlePageMock
    }
};
