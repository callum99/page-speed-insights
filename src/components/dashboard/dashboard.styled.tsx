import styled, {css} from "styled-components";

const FontHeadingDefaults = css`
    font-family: Arial, "Helvetica Neue", sans-serif;
    line-height: 100%;
    font-weight: bold;
`;

const FontRegularDefaults = css`
    font-family: Arial, "Helvetica Neue", sans-serif;
    line-height: 18px;
    font-size: 14px;
`;

export const DashboardContainer = styled.div`
    width: 600px;
    height: 100%;
    background-color: #f1f2f3;
`;

export const DashboardTitle = styled.h1`
    ${FontHeadingDefaults};
    font-size: 30px;
    color: #333333;
    margin-bottom: -10px;
`;

export const DashboardSubTitle = styled.h3`
    ${FontHeadingDefaults};
    font-size:20px;
    color: #666666;
`;

export const DashboardKeyText = styled.span`
    ${FontRegularDefaults};
    color: #000000;
`;

export const DashboardTimeValue = styled.span`
    ${FontRegularDefaults};
    color: red;
`;

export const DomainStyle = styled.span`
    ${FontHeadingDefaults};
    font-size:20px;
    color: #000000;
`;

export const HeaderContainer = styled.div`
    width:100%;
    height:auto;
    display: inline-block;
    background-color:#ffffff;
    padding: 0 20px;
`;

export const BodyContainer = styled.div`
    width:100%;
    height:100%;
    display: inline-block;
`;

export const BodyContentContainer = styled.div`
    width: 100%;
    padding: 0 20px 0 20px;
    display: inline-block;
    height: auto;
`;
