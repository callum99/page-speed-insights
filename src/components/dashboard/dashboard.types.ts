import {currentPageType, historicPagesType} from "../global/global.types";

export interface DashboardProps {
    title: string;
    currentPage?: currentPageType;
    historicPages?: historicPagesType
    hostName?: string;
}
