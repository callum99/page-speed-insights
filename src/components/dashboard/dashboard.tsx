import React from 'react';
import {
    DashboardContainer,
    DashboardTitle,
    DashboardSubTitle,
    DashboardKeyText,
    DashboardTimeValue,
    DomainStyle,
    HeaderContainer,
    BodyContainer,
    BodyContentContainer
} from './dashboard.styled';
import Table from "../table/table";
import {DashboardProps} from "./dashboard.types";

const Dashboard = ({title, currentPage, hostName, historicPages}:DashboardProps) => {
    const TimeToFirstByte = currentPage?.performance.TTFB + 's';
    const FullLoadTime = currentPage?.performance.FullLoadTime + 's';
    const DOMContentLoadedTime = currentPage?.performance.DOMContentLoadedTime + 's';

    return (
        <DashboardContainer>
            <HeaderContainer role="banner">
                <DashboardTitle>{title}</DashboardTitle>
                <DashboardSubTitle>Page Speed metrics for: <DomainStyle>{hostName}</DomainStyle></DashboardSubTitle>
            </HeaderContainer>
            <BodyContainer>
                {!currentPage ? (
                    <BodyContentContainer aria-labelledby="current-performance">
                        <DashboardSubTitle id="current-performance">Waiting on storage data, refresh!</DashboardSubTitle>
                    </BodyContentContainer>
                ) : (
                    <>
                        <BodyContentContainer aria-labelledby="current-performance">
                            <DashboardSubTitle id="current-performance">Current peformance</DashboardSubTitle>
                        </BodyContentContainer>

                        <BodyContentContainer>
                            <DashboardKeyText>DOM Content Load Time:</DashboardKeyText>
                            <DashboardTimeValue>&nbsp;{DOMContentLoadedTime}</DashboardTimeValue>
                            <br/>
                            <DashboardKeyText>Full Load Time:</DashboardKeyText>
                            <DashboardTimeValue>&nbsp;{FullLoadTime}</DashboardTimeValue>
                            <br/>
                            <DashboardKeyText>Time to First Byte:</DashboardKeyText>
                            <DashboardTimeValue>&nbsp;{TimeToFirstByte}</DashboardTimeValue>
                        </BodyContentContainer>
                    </>
                )}

                {historicPages && (
                    <>
                        <BodyContentContainer aria-labelledby="historic-performance">
                            <DashboardSubTitle id="historic-performance">Historic Peformance</DashboardSubTitle>
                        </BodyContentContainer>
                        <Table data={historicPages} />
                    </>)
                }
            </BodyContainer>
        </DashboardContainer>
    );
};

export default Dashboard;