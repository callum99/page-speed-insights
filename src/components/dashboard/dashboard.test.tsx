import renderer from 'react-test-renderer';
import Dashboard from "./dashboard";
import {singlePageMock, historicPageMock} from "./dashboard.const";

it('renders component with required props', () => {
    const tree = renderer.create(<Dashboard title="performance insight" />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders component with all props', () => {
    const tree = renderer.create(<Dashboard title="performance insight" hostName="google.com" currentPage={singlePageMock} historicPages={historicPageMock} />).toJSON();
    expect(tree).toMatchSnapshot();
})