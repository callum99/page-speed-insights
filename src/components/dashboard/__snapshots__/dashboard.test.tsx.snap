// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component with all props 1`] = `
<div
  className="sc-beySPh kUakdV"
>
  <div
    className="sc-gLLuof gzhKmo"
    role="banner"
  >
    <h1
      className="sc-guDLey fAvhev"
    >
      performance insight
    </h1>
    <h3
      className="sc-dmyCSP fIgXkz"
    >
      Page Speed metrics for: 
      <span
        className="sc-jTQCzO boGIVm"
      >
        google.com
      </span>
    </h3>
  </div>
  <div
    className="sc-iBdnpw bOKPZc"
  >
    <div
      aria-labelledby="current-performance"
      className="sc-fsYfdN iFDQyj"
    >
      <h3
        className="sc-dmyCSP fIgXkz"
        id="current-performance"
      >
        Current peformance
      </h3>
    </div>
    <div
      className="sc-fsYfdN iFDQyj"
    >
      <span
        className="sc-hLQSwg etwmzH"
      >
        DOM Content Load Time:
      </span>
      <span
        className="sc-eDLKkx jVPvjI"
      >
         
        0.1s
      </span>
      <br />
      <span
        className="sc-hLQSwg etwmzH"
      >
        Full Load Time:
      </span>
      <span
        className="sc-eDLKkx jVPvjI"
      >
         
        0.1s
      </span>
      <br />
      <span
        className="sc-hLQSwg etwmzH"
      >
        Time to First Byte:
      </span>
      <span
        className="sc-eDLKkx jVPvjI"
      >
         
        0.1s
      </span>
    </div>
    <div
      aria-labelledby="historic-performance"
      className="sc-fsYfdN iFDQyj"
    >
      <h3
        className="sc-dmyCSP fIgXkz"
        id="historic-performance"
      >
        Historic Peformance
      </h3>
    </div>
    <table
      aria-label="Historic Performance"
      className="sc-qZrbh kSekxv"
      role="table"
    >
      <thead
        className="sc-jsEeTM eVVQBk"
        role="rowgroup"
      >
        <tr
          role="row"
        >
          <td
            role="columnheader"
          >
            Date
          </td>
          <td
            role="columnheader"
          >
            DOMContentLoadTime
          </td>
          <td
            role="columnheader"
          >
            FullLoadTime
          </td>
          <td
            role="columnheader"
          >
            TTFB
          </td>
        </tr>
      </thead>
      <tbody
        role="rowgroup"
      >
        <tr
          className="sc-irLvIq ksyBsF"
          role="row"
        >
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            2024-04-24T10:52:20.779Z
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
        </tr>
        <tr
          className="sc-irLvIq ksyBsF"
          role="row"
        >
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            2024-04-24T10:12:09.664Z
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
        </tr>
        <tr
          className="sc-irLvIq ksyBsF"
          role="row"
        >
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            2024-04-24T09:34:38.019Z
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
        </tr>
        <tr
          className="sc-irLvIq ksyBsF"
          role="row"
        >
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            2024-04-24T09:10:46.905Z
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
        </tr>
        <tr
          className="sc-irLvIq ksyBsF"
          role="row"
        >
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            2024-04-24T08:13:14.099Z
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
          <td
            className="sc-kFCroH flRlee"
            role="cell"
          >
            0.1
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`renders component with required props 1`] = `
<div
  className="sc-beySPh kUakdV"
>
  <div
    className="sc-gLLuof gzhKmo"
    role="banner"
  >
    <h1
      className="sc-guDLey fAvhev"
    >
      performance insight
    </h1>
    <h3
      className="sc-dmyCSP fIgXkz"
    >
      Page Speed metrics for: 
      <span
        className="sc-jTQCzO boGIVm"
      />
    </h3>
  </div>
  <div
    className="sc-iBdnpw bOKPZc"
  >
    <div
      aria-labelledby="current-performance"
      className="sc-fsYfdN iFDQyj"
    >
      <h3
        className="sc-dmyCSP fIgXkz"
        id="current-performance"
      >
        Waiting on storage data, refresh!
      </h3>
    </div>
  </div>
</div>
`;
